activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -84,6 +84,20 @@ def sql_color(sql)
       def logger
         ActiveRecord::Base.logger
       end
+
+      def debug(progname = nil, &block)
+        return unless super
+
+        backtrace = Rails.backtrace_cleaner.clean caller
+
+        relevant_caller_line = backtrace.detect do |caller_line|
+          !caller_line.include?('/initializers/')
```

Comment:
Welp @sgrif, turns out this line was what made it work. I can only surmise that it somehow eliminates everything except the non-framework caller lines, but I honestly don't know yet.